src/Stack.php

```diff
@@ -9,16 +9,23 @@
 class Stack
 {
     /**
+     * @var callable
+     */
+    private $default;
+
+    /**
      * @var array
      */
     private $middleware = [];
 
     /**
+     * @param callable $default to call when no middleware is available
      * @param array $middleware
      */
-    public function __construct(array $middleware = [])
+    public function __construct(callable $default, ...$middleware)
     {
-        array_map([$this, 'append'], $middleware);
```

Comment:
Yeah, hhvm does not support `public function __construct(callable $default, ServerMiddlewareInterface ...$middleware)` :confused: